src/tests/test_magic_cmd.py

```diff
@@ -92,8 +92,8 @@ def ip_snippets(ip):
         ],
         [
             "%sqlcmd columns",
-            UsageError,
-            "the following arguments are required: -t/--table",
+            SystemExit,
+            "2",
```

Comment:
What's this error? why did it change?

Reply:
> I don't think this is used, right?

We earlier had a custom parser called `CmdParser` where we were overriding the the exit method of the `argparse.ArgumentParser()` to throw UsageError.  Now we have removed it, so the when there is a command with incomplete arguments (eg. table/schema is missing) `argparse.ArgumentParser()` throws a SystemExit. But our `_validate_execute_inputs` prints the issue message to the users.

Reply:
ah, this was a mistake on my end.

I suggested to remove CmdParser because I thought we were not using the `.error` method because I never saw any calls to it. but now I remembered that it'll get called when there is some argument validation error.


@AnirudhVIyer: please re-add it as it was. and add a docstring saying that we're subclassing ArgumentParser because Argument parser will raise a SysExit error when encountering argument validation errors, but in our cases, we want to raise a UsageError